scenes/cslib_object.txt

```diff
@@ -70,6 +79,46 @@
 *gosub _set_main_instance p_objectname p_index
 *return
 
+*comment GET_BY_FIELD
+*comment ------------------------------------------------------------------
+*comment Takes the specified field name and field value and finds the
+*comment first instance of a match that it findscl and copies that data
+*comment into the main instance
+*comment If no match is found, all the fields of the main object are set
+*comment to false
+*comment ------------------------------------------------------------------
+*comment
+*comment params:
+*comment	p_objectname (string): the name of the object
+*comment	p_field (string): field to match
+*comment	p_value (string): field to match
+*comment returns:
+*comment	none
+*comment usage example:
+*comment 	*gosub_scene cslib_object get "dragon" "name" "Wally"
+*comment ------------------------------------------------------------------
+
+*label get_by_field
+*params p_object p_field p_value
+*set cslib_ret ""
+*temp foundit false
+*temp n 1
+*temp array_count {p_object&"_max"}
```

Comment:
By committing the we are agreeing to hold variables to array standards in ChoiceScript.
I'm fine with this, but @diegoboston, weren't you hesitant on the last PR?

Reply:
We should probably use _count instead of _max here (see: https://github.com/ChoicescriptIDE/cslib/wiki/Arrays).

Reply:
OK